### DISCUSSION TITLE
Set specific element size for one part

### DISCUSSION THREAD

**Discussion by Maschinenausleger** [Eomys/pyleecan#633] [Q&A] [answered]
Hi, everybody. 
I am new to pyleecan and I have a new project where I need to adapt element size for a specific part of the machine (Rotor). I saw in the tutorial the opportunity to specify the air gap element size via the femm_dict_enforced. I assume that this should also work for the rotor. unfortunately I can't figure out how to access the rotor. Has someone an idea how to solve this problem?

Thanks in advance 
Chris

**Comment by BonneelP** [comment, marked as answer]
Hello,

Thank you for your interest in pyleecan. The mesh size of each part is computed in: pyleecan\Functions\FEMM\comp_FEMM_dict.py
When running a simulation, the full FEMM_dict is saved in out.mag.internal.FEMM_dict. So you can run a simulation with Nt_tot=1, get the generated FEMM_dict and check the content to find which key you need to set. I assume it will be something like FEMM_dict["mesh"]["Rotor-0"]["meshsize_magnet"] for instance. 

On a side note you can also use simu.mag.Kmeshh_finess to change the mesh size for all the machine.
![image](https://github.com/Eomys/pyleecan/assets/20533034/20f93dba-6705-4cd7-bfe7-7c0f1138ba00)

Best regards,
Pierre

**Reply by Maschinenausleger** [reply]
Hi, thanks for your quick response. 
I was exactly what I was looking for.
